### PR TITLE
Introduce pinned env in noise protocol

### PR DESCRIPTION
**Pull request by velzevur**
PT [SC noise: add pinned env hash](https://www.pivotaltracker.com/story/show/165593709)

This PR enriches off-chain communication with pinned env for:
* new off-chain updates so the `state_hash` can be computed based on a certain pinned environment as well as checking the initial upater's signatures
* new off-chain change confirmation ( `_ACK`) for second participant's signatures checks

Signatures checks shall be checked according to the pinned environment esp with regard of GAs

This PR replaces #2368

### CODE REVIEW

File: apps/aechannel/src/aesc_codec.erl [hunk 1]
```diff
@@ -233,43 +233,53 @@ dec_ch_reest_ack(<< ChainHash:32/binary
...
     Length = byte_size(Data),
     << ?ID_FND_CREATED:1 /unit:8
      , ChanId         :32/binary
+     , BlockHash      :32/binary
      , Length         :2 /unit:8
      , Data           :Length/bytes >>.
 
 -spec dec_fnd_created(binary()) -> fnd_created_msg().
 dec_fnd_created(<< ChanId:32/binary
+                 , BlockHash:32/binary
```
Comment (lucafavatella) [inline apps/aechannel/src/aesc_codec.erl]: Doc: https://github.com/aeternity/protocol/blob/56793c15061a1b316ccb7506cea1daf2daeb3bd2/channels/OFF-CHAIN.md#funding_created

Doc needs update.

Versioning for incompatibility?

Comment (lucafavatella) [reply apps/aechannel/src/aesc_codec.erl]: In other points too.

**Comment by lucafavatella** [comment]
I took the liberty of labelling as WIP - I understand planned updates.

File: apps/aechannel/src/aesc_codec.erl (hunk 1 above)
Comment (velzevur) [reply apps/aechannel/src/aesc_codec.erl]: Addressed in aeternity/protocol#343

File: docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise [hunk 2]
```diff
@@ -0,0 +1,2 @@
+* Introduces a pinned environment in State Channels noise protocol. This is
+  not backwards compatible
```
Comment (lucafavatella) [inline docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise]: If I understand this correctly:
* The noise-based state channel protocol has no versioning. ([Past draft](https://github.com/aeternity/aeternity/pull/2204).)
* New node and old node manage to connect at TCP and noise level, though when exchanging any meaningful message either one of them (depending on who requested whom what first) close the TCP connection.

Right?

File: docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise [hunk 3]
```diff
@@ -0,0 +1,2 @@
+* Introduces a pinned environment in State Channels noise protocol. This is
```
Comment (lucafavatella) [inline docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise]: So there is no versioning in the actual channel protocol. Plus there is no versioning [in the doc](https://github.com/aeternity/protocol/pull/343) - apart from the tag name of course. May I suggest using these release notes for detailing what changed in the protocol?
```suggestion
* Introduces a pinned environment in State Channels noise protocol (off-chain), by introducing a `block_hash` field in some of the messages. This is
```

File: docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise (hunk 2 above)
Comment (velzevur) [reply docs/release-notes/next-fortuna/PT-165593709_pinned_env_in_noise]: Yes

**Pull request merged**